### PR TITLE
Adding PHPUnit tests.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: tests/src/Plugin/LibrariesCDN/cdnjsTest.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace Drupal\Tests\libraries_cdn\Plugin\LibrariesCDN;
+use Drupal\libraries_cdn\Plugin\LibrariesCDN\CDNJS;
+use Drupal\libraries_cdn\Type\CDNBaseInterface;
+use Drupal\service_container\Legacy\Drupal7;
+
+/**
+ * @coversDefaultClass \Drupal\libraries_cdn\Plugin\LibrariesCDN\cdnjs
+ */
+class cdnjsTest extends \PHPUnit_Framework_TestCase {
```
Comment (LionsAd) [inline tests/src/Plugin/LibrariesCDN/cdnjsTest.php]: CDNJSTest or CdnJsTest

File: tests/src/Plugin/LibrariesCDN/cdnjsTest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Contains \Drupal\Tests\Libraries_cdn\Plugin\LibrariesCDN\cdnjsTest.
+ */
+
+namespace Drupal\Tests\libraries_cdn\Plugin\LibrariesCDN;
+use Drupal\libraries_cdn\Plugin\LibrariesCDN\CDNJS;
+use Drupal\libraries_cdn\Type\CDNBaseInterface;
+use Drupal\service_container\Legacy\Drupal7;
+
+/**
+ * @coversDefaultClass \Drupal\libraries_cdn\Plugin\LibrariesCDN\cdnjs
```
Comment (LionsAd) [inline tests/src/Plugin/LibrariesCDN/cdnjsTest.php]: cdnjs => CDNJS

File: tests/src/Plugin/LibrariesCDN/cdnjsTest.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  /* @var CDNBaseInterface $this->plugin */
+  protected $plugin;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $drupal7 = \Mockery::mock('\Drupal\service_container\Legacy\Drupal7');
```
Comment (LionsAd) [inline tests/src/Plugin/LibrariesCDN/cdnjsTest.php]: You need to use $this->drupal7 = ...

File: tests/src/Plugin/LibrariesCDN/cdnjsTest.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    $this->plugin = new CDNJS(array(), 'cdnjs', array(), $drupal7);
+  }
+
+  /**
+   * @covers ::setLibrary()
+   * @covers ::getLibrary()
+   */
+  public function test_setgetLibrary() {
+    $this->plugin->setLibrary('jquery');
```
Comment (LionsAd) [inline tests/src/Plugin/LibrariesCDN/cdnjsTest.php]: You need to do something like:

$this->drupal7->shouldReceive('drupal_http_request')->with(...)->andReturn(...);

See our drupal_http_message unit tests in ServiceContainer itself.

e.g. The mock does not know how to interpret the drupal_http_request calls it is getting.

You need to specify what the behavior needs to be.

Comment (LionsAd) [inline tests/src/Plugin/LibrariesCDN/cdnjsTest.php]: Also see:

http://docs.mockery.io/en/latest/reference/expectations.html

File: tests/src/Plugin/LibrariesCDN/cdnjsTest.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    $this->assertEquals('jquery', $this->plugin->getLibrary());
+  }
+
+  /**
+   * @covers ::isAvailable()
+   */
+  public function test_isAvailable() {
+    $this->plugin->setLibrary('jquery');
```
Comment (LionsAd) [inline tests/src/Plugin/LibrariesCDN/cdnjsTest.php]: The same here, but with different rules:

$this->drupal7->shouldReceive('drupal_http_request')->once()->with(...)->andReturn(...);

**Comment by drupol** [comment]
pol@x230 ~/g/l/tests> ./run-tests.sh
PHPUnit 3.7.38 by Sebastian Bergmann.

Configuration read from /home/pol/git/libraries_cdn/tests/phpunit.xml.dist

.E

Time: 44 ms, Memory: 5.00Mb

There was 1 error:

1) Drupal\Tests\libraries_cdn\Plugin\LibrariesCDN\CDNJSTest::test_isAvailable
Undefined index: code

/home/pol/git/libraries_cdn/src/Type/CDNBase.php:120
/home/pol/git/libraries_cdn/tests/src/Plugin/LibrariesCDN/CDNJSTest.php:56

FAILURES!
Tests: 2, Assertions: 2, Errors: 1.
pol@x230 ~/g/l/tests>

File: tests/src/Plugin/LibrariesCDN/CDNJSTest.php [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+  /**
+   * @covers ::query()
+   */
+  public function test_query() {
+    $this->plugin->setLibrary('jquery');
+    $configuration = $this->plugin->getConfiguration();
+
+    $this->drupal7->shouldReceive('drupal_http_request')
+      ->with(sprintf($configuration['urls']['getInformation'], $this->plugin->getLibrary()), array())
+      ->andReturn(array('code' => 200, 'data' => '{"name":"jquery"}'));
```
Comment (LionsAd) [inline tests/src/Plugin/LibrariesCDN/CDNJSTest.php]:
```
$this->drupal7
  ->shouldReceive('drupal_http_request')
  ->with($url, array())
  ->andReturn(array('code' => 200, 'data' => json_encode(array('...'))));
```

OR

$this->drupal7
  ->shouldReceive('drupal_http_request')
  ->with($url, array())
  ->andReturn(array('code' => 404));
```

Comment (LionsAd) [inline tests/src/Plugin/LibrariesCDN/CDNJSTest.php]: oh, lol, my telepathic transmission did reach you first :D.

**Pull request merged**